low-level/framework/utils/storage_encl.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+
+# CORTX Python common library.
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import time
+import json
+
+from framework.base import sspl_constants as sspl_const
+from framework.platforms.realstor.realstor_enclosure \
+    import singleton_realstorencl
+
+
+class StorageEncl:
+    """Base class for storage enclosure related utility functions."""
+    def get_enclosure_logs(self, file_name, logger):
+        """Accumulate enclosure logs & write to supplied file."""
+        url = singleton_realstorencl.build_url(
+            singleton_realstorencl.URI_CLIAPI_BASE)
+        COLLECTING_DEBUG_LOG_STARTED = False
+        for encl_trigger_log_retry_index in range(0,
+                                sspl_const.ENCL_TRIGGER_LOG_MAX_RETRY):
+            post_data_string = '{0}/"{1}"{2}"{3}'.format(
+                singleton_realstorencl.URL_ENCLLOGS_POSTDATA,
+                sspl_const.SUPPORT_REQUESTOR_NAME,
+                sspl_const.SUPPORT_EMAIL_ID,
+                sspl_const.SUPPORT_CONTACT_NUMBER)
+            response = singleton_realstorencl\
+                .ws_request(url,
+                            singleton_realstorencl.ws.HTTP_POST,
+                            post_data=post_data_string)
+
+            if not response:
+                logger.error("{0} status unavailable as ws request {1}"
+                             " failed".format(
+                                 singleton_realstorencl.LDR_R1_ENCL,
+                                 url))
+                break
```

Comment:
Don't we need to retry here?

Reply:
If response is empty, then we are not retrying but if it is not None but also not expected then we are retrying in upper loop.

Reply:
Done